src/cider/nrepl/middleware/apropos.clj

```diff
@@ -17,17 +18,26 @@
 ;; abbreviated version (i.e. first sentence only) may be returned for
 ;; symbol-only searches.
 
+(def special-forms
+  "Special forms that can be apropo'ed."
+  (concat (keys (var-get #'clojure.repl/special-doc-map))
+          '[& catch finally]))
+
 (defn var-name
-  "Return a var's namespace-qualified name as a string."
+  "Return a special form's name or var's namespace-qualified name as a string."
   [v]
-  (str/join "/" ((juxt (comp ns-name :ns) :name)
-                 (meta v))))
+  (if (special-symbol? v)
+    (str (:name (info/resolve-special v)))
```

Comment:
So all these functions conflate symbols and vars now? 

If that's the way, then I think instead of inserting `(special-symbol? ...)` checks everywhere it's better to have all these internals in one place - `(var-meta ..)`.  Also IMO including one middleware into another is not ideal. All these small utilities could easily live in a separate utility namespace (precisely what has been done as an aside of #405 btw).

@xiongtx do you plan for more imediate changes? Could you please give me a day or so to merge #405 as any changes to info and apropos will likely introduce hard conflicts with that branch. Thanks.